pkg/cmd/cli/cmd/set/buildhook.go

```diff
@@ -139,7 +139,7 @@ func (o *BuildHookOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, ar
 	o.Cmd = cmd
 
 	mapper, typer := f.Object()
-	o.Builder = resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), kapi.Codecs.UniversalDecoder()).
+	o.Builder = resource.NewBuilder(mapper, f.CategoryExpander(), typer, resource.ClientMapperFunc(f.ClientForMapping), kapi.Codecs.UniversalDecoder()).
```

Comment:
hm, I believe the `NewBuilder` func added to the [factory_builder](https://github.com/kubernetes/kubernetes/blob/release-1.7/pkg/kubectl/cmd/util/factory_builder.go#L154) should be in [kube 1.7](https://github.com/kubernetes/kubernetes/blob/release-1.7/pkg/kubectl/cmd/set/set_resources.go#L144).

Any reason why we are still using `resource.NewBuilder`?